packages/service-library/src/servicelib/monitor_slow_callbacks.py

```diff
@@ -19,13 +21,22 @@ def enable(slow_duration_secs: float, incidents: List[SlowCallback]) -> None:
     _run = asyncio.events.Handle._run
 
     def instrumented(self):
-        t0 = time.monotonic()
-        retval = _run(self)
-        dt = time.monotonic() - t0
-        if dt >= slow_duration_secs:
-            task_info = _format_handle(self)
-            incidents.append(SlowCallback(msg=task_info, delay_secs=dt))
-            logger.warning("Executing %s took %.3f seconds", task_info, dt)
+        # unsetting profiler, helps with development mode and tests
+        sys.setprofile(None)
+
+        with Profiler(interval=slow_duration_secs) as profiler:
+            t0 = time.monotonic()
+
+            retval = _run(self)
+
+            dt = time.monotonic() - t0
+
+        profiler_result = profiler.output_text(unicode=True, color=False, show_all=True)
+        slow_callbacks_detected = "No samples were recorded." not in profiler_result
```

Comment:
Nonetheless, I find this a bit hacky ... 
Does the profiler have a flag that tells whether it was triggered? or alternatively, why not keep the condition we had before? 

```python
with Profiler() as profile:
    ... # code
   dt = ...

if dt >= slow_duration_secs:
    # profiler_result ...
```

Reply:
I've already checked and sadly this looked like the best option to check for that string inside the message.

Actually I've also thought about keeping the old check but did not like it. This was the simplest result without spending too much time on it.

Reply:
I had the same feeling at first, but:
- the documentation clearly states `If you get "No samples were recorded." because your code executed in under 1ms, hooray!`, [see here](https://github.com/joerick/pyinstrument#profile-a-specific-chunk-of-code)
- there are tests which cover the fact that the check works; you did a really good job with them!
- lastly, if there is a delay between the two separate ways of counting time, we could register an inexistent long lasting handler. The string searching avoids this issue.

Reply:
not fully convinced ...